integrated_channels/integrated_channel/management/commands/transmit_learner_data.py

```diff
@@ -0,0 +1,96 @@
+    Management command which transmits learner course completion data to the IntegratedChannel(s) configured for the
+    given EnterpriseCustomer.
+
+    Collect the enterprise learner data for enrollments with data sharing consent, and transmit each to the
+    EnterpriseCustomer's configured IntegratedChannel(s).
+    """
+    help = """
+    Transmit enterprise learner course completion data for the given enterprise customer.
+    """
+    integrated_channel_choices = {
+        'sap': SAPSuccessFactorsEnterpriseCustomerConfiguration,
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--enterprise_customer',
+            dest='enterprise_customer',
+            default=None,
+            metavar='ENTERPRISE_CUSTOMER_UUID',
+            help='Transmit data for this EnterpriseCustomer to its configured integration channel(s).',
+        )
+        parser.add_argument(
+            '--channel',
+            dest='channel',
+            default=None,
+            metavar='INTEGRATED_CHANNEL',
+            help='Transmit data to this INTEGRATED_CHANNEL.  '
+            'Omit option to transmit to all configured, active integration channels.',
+            choices=self.integrated_channel_choices.keys(),
+        )
+
+    def handle(self, *args, **options):
+        """
+        Collect the learner data for the given EnterpriseCustomer, and transmit it to its active integration channels.
+        """
+        # Ensure there's a valid enterprise customer
+        enterprise_customer_uuid = options['enterprise_customer']
+        if enterprise_customer_uuid:
+            try:
+                enterprise_customer = EnterpriseCustomer.active_customers.get(uuid=enterprise_customer_uuid)
+            except EnterpriseCustomer.DoesNotExist:
+                raise CommandError('Enterprise customer {} not found, or not active'.format(enterprise_customer_uuid))
+        else:
+            enterprise_customer = None
+
+        channel_code = options['channel']
+        if channel_code is not None:
+            if channel_code not in self.integrated_channel_choices:
+                raise CommandError('Invalid integrated channel: {}'.format(channel_code))
+            plugin_classes = [self.integrated_channel_choices.get(channel_code)]
+        else:
+            # Otherwise, use all the configured channel types
+            plugin_classes = self.integrated_channel_choices.values()
+
+        active_integrated_channel_found = False
+        for plugin_class in plugin_classes:
+            integrated_channels = plugin_class.objects.filter(active=True)
+            if enterprise_customer:
+                integrated_channels.filter(enterprise_customer=enterprise_customer)
+
+            # Transmit the learner data to each integrated channel
+            for integrated_channel in integrated_channels:
+                self.transmit_learner_data(integrated_channel)
+                active_integrated_channel_found = True
```

Comment:
Leaving that up to you guys..  do you need the jenkins job to fail if no data gets sent?